### ISSUE TITLE
Task 2 (Dance Performance Scoring) implementation missing — no Gaussian distribution head, no uncertainty-weighted loss, no ground truth labels

### ISSUE THREAD

**Issue by iamhenryhuang** [YunZhongNikki/DanceMVP#1]
Hi, I've been trying to reproduce Task 2 (Dance Performance Scoring) as described in the paper, and I can't find any corresponding implementation in this repository.

The paper formulates this task as predicting a Gaussian score distribution (mean μ and variance σ²) with an uncertainty-weighted negative log-likelihood loss:

L2 = (1/N) Σ [ (α / σ(xi)²) ||yi − μ(xi)||² + β log σ(xi)² ]
Going through the code, here's what I found:

No loss function matches this formula anywhere in the repo. The only loss used in test_our_text_prompt.py is torch.nn.CrossEntropyLoss() (line 174), applied for classification. There's no α/β weighting, no log σ² term, and no Gaussian NLL computation anywhere in the codebase.

No model head outputs a (μ, σ) pair. model_mlp (used at line 210) outputs a single features tensor that's directly used for CLIP-style contrastive similarity against text embeddings (logits = features @ text_out.t()) — there's no branch producing separate mean/variance outputs.

There are two suspicious dead variables in test_our_text_prompt.py:

pred_scores = []   # line 182
sigma = []         # line 183
Both are declared right before the training loop but are never appended to or read anywhere in the script. The naming (pred_scores, sigma) strongly suggests these were meant to hold Task 2's predicted score and predicted variance, but the implementation connecting them to the model/loss was never completed or never released.

Separately, I also don't see the expert-labeled performance score ground truth anywhere in the dataset (no CSV/JSON/label file in either this repo or ImperialDance-Dataset) — so even if the model code were present, there'd be no labels to train against.

Could you clarify:

Is the Task 2 training code (Gaussian distribution head + uncertainty-weighted loss) available in a different branch/version, or was it never included in this release?
Are the expert performance score labels available separately, or only used internally for the paper's reported results?
Reproducing the Dance Performance Scoring results currently isn't possible with what's in this repo — any pointers would help a lot. Thanks!

**Comment by iamhenryhuang** [comment]
I also went through the primitive segmentation code and it doesn't match the beat-based segmentation described for the method (detect musical beats via librosa, treat every 8 beats as one primitive, and segment motion using those same boundaries).

What seg_motion() actually does in both data_aug/seg_dataset.py (lines 239-247) and data_aug/seg_dataset_text.py (lines 263-271) is a fixed time-based split, unrelated to beat detection:


def seg_motion(motion_arry):
    seg_arr[i, 0, :, :] = motion_arry[i, 100:400, :, :]   # seconds 1-4
    seg_arr[i, 1, :, :] = motion_arry[i, 400:700, :, :]   # seconds 4-7
    seg_arr[i, 2, :, :] = motion_arry[i, 700:1000, :, :]  # seconds 7-10
It just drops the first 1 second (100 frames) as a buffer, then splits the remaining 9 seconds into 3 fixed 300-frame (3-second) chunks — always exactly 3 primitives per 10-second clip, regardless of tempo or actual beat positions.

Separately, librosa.beat.beat_track is used in the repo (cal_hitrate() in both files), but only for the Task 3 BAR computation — comparing beat timestamps against motion-intensity peaks. It's never called from or connected to seg_motion(), so beat detection has no role in primitive boundary segmentation anywhere in this codebase.

Could you clarify whether the 8-beats-per-primitive segmentation was only used to build the released .npy motion files offline (i.e., pre-segmentation happened before this repo's code runs), or whether the fixed 3-way split in seg_motion() is the actual segmentation method and the beat-based description refers to something else? Reproducing primitive boundaries that align with musical structure isn't possible with the current seg_motion() implementation as released. Thanks!

**Comment by YunZhongNikki** [comment]
Thanks for pointing this out. To clarify, this repository contains an earlier/testing implementation and does not represent the complete implementation used for all experiments reported in the paper.

Importantly, the released scripts in this repository were provided to reproduce Task 1 only. They do not contain the implementations, annotations, or outputs for Tasks 2 and 3. The Task 2 and Task 3 annotations have also been used in our subsequent work and are not currently publicly available. Therefore, the Task 1 scripts in this repository should not be interpreted as implementations of Tasks 2 or 3.

Regarding the primitive segmentation, the beat-based segmentation described in the paper, including the 8-beats-per-primitive procedure, was implemented in the version released through [this website](www.imperial.ac.uk/personal-robotics/software), as referenced in the paper. The fixed segmentation in seg_motion() in this repository is part of the testing implementation and should not be interpreted as the segmentation procedure used for the reported results.

As this repository is no longer actively maintained and does not represent the final paper implementation, it will be archived for reference. I hope this clarifies both the scope of this repository and the distinction between the released Task 1 scripts and the other tasks described in the paper. Thanks for your understanding.
